saq/queue.py

```diff
@@ -505,8 +507,12 @@ async def map(
             }
             for kw in iter_kwargs
         ]
-        job_ids = [Job.id_from_key(key["key"]) for key in iter_kwargs]
-        pending_job_ids = set(job_ids)
+        job_keys = [key["key"] for key in iter_kwargs]
+        pending_job_keys = set(job_keys)
+        pending_job_ids = set(
+            Job.id_from_key(job_key, queue_name=self.name)
+            for job_key in pending_job_keys
+        )
 
         async def callback(job_id, status):
```

Comment:
If `listen` uses job keys as a parameter, it should pass job keys to the callback. In other words, I think we should extract the key out of the id before calling this callback. That would make this `map` method simpler.